clang-tools-extra/clang-tidy/cppcoreguidelines/MissingStdForwardCheck.cpp

```diff
@@ -125,12 +127,13 @@ void MissingStdForwardCheck::registerMatchers(MatchFinder *Finder) {
                    hasAncestor(expr(hasUnevaluatedContext())))));
 
   Finder->addMatcher(
-      parmVarDecl(parmVarDecl().bind("param"), isTemplateTypeParameter(),
-                  hasAncestor(functionDecl().bind("func")),
-                  hasAncestor(functionDecl(
-                      isDefinition(), equalsBoundNode("func"), ToParam,
-                      unless(anyOf(isDeleted(), hasDescendant(std::move(
-                                                    ForwardCallMatcher))))))),
+      parmVarDecl(
+          parmVarDecl().bind("param"), hasAnyName(), isTemplateTypeParameter(),
```

Comment:
You should also ignore unused variables, like `[[maybe_unused]]` with:
```
unless(hasAttr(attr::Kind::Unused)),
```

And you could also use:
```
unless(isReferenced()),

with:

AST_MATCHER(VarDecl, isReferenced) { return Node.isReferenced(); }
```

Reply:
I went with `unless(hasAttr(attr::Kind::Unused))` and `hasIdentifier`. `isReferenced()` doesn't play to nicely with constructor initializer lists. It seems that it doesn't consider a declaration used when it's only used to initialize a field.